static/events.ts

```diff
@@ -0,0 +1,151 @@
+    initialised: () => void;
+    irViewClosed: (compilerId: number) => void;
+    irViewOpened: (compilerId: number) => void;
+    languageChange: (editorId: number, newLangId: string, treeId?: boolean | number) => void;
+    modifySettings: (modifiedSettings: Partial<SiteSettings>) => void;
+    // TODO: Add type specificity
+    motd: (data: unknown) => void;
+    newSource: (editorId: number, newSource: string) => void;
+    optViewClosed: (compilerId: number) => void;
+    optViewOpened: (compilerId: number) => void;
+    outputClosed: (compilerId: number) => void;
+    outputOpened: (compilerId: number) => void;
+    panesLinkLine: (
+        compilerId: number,
+        lineNumber: number,
+        colBegin: number,
+        colEnd: number,
+        reveal: boolean,
+        sender: string
+    ) => void;
+    ppViewClosed: (compilerId: number) => void;
+    ppViewOpened: (compilerId: number) => void;
+    // TODO: See if this is necessary. Nothing emits this event
+    ppViewOptionsUpdated: (compilerId: number, options: string, recompile: boolean) => void;
+    requestCompilation: (editorId: number, treeId: boolean | number) => void;
+    requestMotd: () => void;
+    requestSettings: () => void;
+    resendCompilation: (compilerId: number) => void;
+    requestCompiler: (compilerId: number) => void;
+    // TODO: Bug? There are no listeners for this event, only Pane.ts emits it
+    requestFilters: (compilerId: number) => void;
+    resendExecution: (executorId: number) => void;
+    resize: () => void;
+    rustHirViewClosed: (compilerId: number) => void;
+    rustHirViewOpened: (compilerId: number) => void;
+    rustMacroExpViewClosed: (compilerId: number) => void;
+    rustMacroExpViewOpened: (compilerId: number) => void;
+    rustMirViewClosed: (compilerId: number) => void;
+    rustMirViewOpened: (compilerId: number) => void;
+    // TODO: There are no emitters for this event
+    selectLine: (editorId: number, lineNumber: number) => void;
+    settingsChange: (newSettings: SiteSettings) => void;
+    setToolInput: (compilerId: number, toolId: number, string: string) => void;
+    shown: () => void;
+    themeChange: (newTheme: Theme | null) => void;
+    toolClosed: (compilerId: number, toolState: unknown) => void;
+    toolInputChange: (compilerId: number, toolId: number, input: string) => void;
+    toolInputViewClosed: (compilerId: number, toolId: number, input: string) => void;
+    toolInputViewCloseRequest: (compilerId: number, toolId: number) => void;
+    // TODO: Only emitted
+    toolInputViewOpened: (toolId: number) => void;
+    toolOpened: (compilerId: number, toolState: unknown) => void;
+    toolSettingsChange: (compilerId: number) => void;
+    treeClose: (treeId: number) => void;
+    treeCompilerEditorExcludeChange: (treeId: number, compilerId: number, editorId: number) => void;
+    treeCompilerEditorIncludeChange: (treeId: number, compilerId: number, editorId: number) => void;
+    // Todo: Only emitted
```

Comment:
Listened here https://github.com/compiler-explorer/compiler-explorer/blob/335f683b6b065072061496dd63e31201bece108d/static/hub.ts#L157-L163

Reply:
Ohh. I know why I couldn't find any of this events. I grepped in single-line mode

Reply:
This means I might have missed some events! But hopefuly they show up as errors if not present